### PR TITLE
add code capture teacher learner performance data via ussd javascript

### PR DESCRIPTION
**Pull request by gitumarkk**
### User Story

so that learner and teacher performance data can be analysed as a district admin I can  capture performance data

### CODE REVIEW

**Comment by gitumarkk** [comment]
@imsickofmaps ready for review.

File: js_sandbox/lib/go-rts-zambia.js [hunk 1]
```diff
@@ -223,6 +255,32 @@ function GoRtsZambia() {
...
+            "pupils_books_number": im.get_user_answer('perf_teacher_pupils_books_number'),
+            "reading_lesson": im.get_user_answer('perf_teacher_reading_lesson'),
+            "pupil_engagement_score": im.get_user_answer('perf_teacher_pupil_engagement_score'),
+            "attitudes_and_beliefs": im.get_user_answer('perf_teacher_attitudes_and_beliefs'),
+            "training_subtotal": im.get_user_answer('perf_teacher_training_subtotal'),
+            "academic_level": "/api/v1/data/achievement/" + im.get_user_answer('perf_teacher_academic_level') + "/",
+            "reading_assessment": im.get_user_answer('perf_teacher_reading_assessment'),
+            "reading_total": im.get_user_answer('perf_teacher_reading_total'),
+            "emis": "/api/v1/school/emis/" + emis + "/",
+            "created_by_da": "/api/v1/district_admin/" + id + "/"
```
Comment (imsickofmaps) [inline js_sandbox/lib/go-rts-zambia.js]: Will DRY it up and appended created_by's

File: js_sandbox/lib/go-rts-zambia.js [hunk 2]
```diff
@@ -257,6 +315,40 @@ function GoRtsZambia() {
...
+            "phonetic_awareness": im.get_user_answer('perf_learner_boys_phonetic_awareness'),
+            "vocabulary": im.get_user_answer('perf_learner_boys_vocabulary'),
+            "reading_comprehension": im.get_user_answer('perf_learner_boys_reading_comprehension'),
+            "writing_diction": im.get_user_answer('perf_learner_boys_writing_diction'),
+            "outstanding_results": im.get_user_answer('perf_learner_boys_outstanding_results'),
+            "desirable_results": im.get_user_answer('perf_learner_boys_desirable_results'),
+            "minimum_results": im.get_user_answer('perf_learner_boys_minimum_results'),
+            "below_minimum_results": im.get_user_answer('perf_learner_boys_below_minimum_results'),
+            "emis": "/api/v1/school/emis/" + emis + "/",
+            "created_by_da": "/api/v1/district_admin/" + id + "/"
```
Comment (imsickofmaps) [inline js_sandbox/lib/go-rts-zambia.js]: Same on DRY it up

File: js_sandbox/lib/go-rts-zambia.js [hunk 3]
```diff
@@ -638,6 +750,26 @@ function GoRtsZambia() {
...
+            return self.add_emis_district_official_to_contacts(content);
+        },
+        "The emis does not exist, please try again. This should have 4-6 digits e.g 4351."
+    ));
+
+    self.add_state(new FreeText(
+        "add_emis_perf_learner_boys_total",
+        "perf_learner_boys_total",
+        "Please enter the school's EMIS number that you would like to report on. This should have 4-6 digits e.g 4351.",
+        function(content){
```
Comment (imsickofmaps) [inline js_sandbox/lib/go-rts-zambia.js:731]: Not sure it's safe running callbacks in a check? Should make this is check only and save on_exit I think...

Comment (imsickofmaps) [inline js_sandbox/lib/go-rts-zambia.js:731]: Not safe, will remove

**Comment by imsickofmaps** [comment]
@smn ready for review

File: js_sandbox/lib/go-rts-zambia.js [hunk 4]
```diff
@@ -216,14 +257,27 @@ function GoRtsZambia() {
...
         };
 
         return data;
     };
 
-    self.performance_data_learner_collect = function(emis, id){
+    self.performance_data_learner_collect_by_head = function(emis, id){
+        var data = self.performance_data_learner_collect(emis);
+        data[0]["created_by"] = "/api/v1/data/headteacher/" + id + "/";
+        data[1]["created_by"] = "/api/v1/data/headteacher/" + id + "/";
```
Comment (smn) [inline js_sandbox/lib/go-rts-zambia.js:269]: this index thing is going to get confusing after a while.
Can `performance_data_learner_collect` not return an `{boys: {}, girls: {}}` object so you can refer to them by name?

File: js_sandbox/lib/go-rts-zambia.js [hunk 5]
```diff
@@ -543,21 +607,29 @@ function GoRtsZambia() {
                     ]
                 );
             } else {
-                // recognised user
                 if (im.config.performance_monitoring_active){ // Performance monitoring active
+                    // recognised user
+                    if (result.contact["extras-rts_district_official_district_id"] === undefined) {
```
Comment (smn) [inline js_sandbox/lib/go-rts-zambia.js:612]: Is it possible to log this somewhere? I'm worried that if somehow someone enters the contacts database (say via CSV import) then all of a sudden the user's a head teacher.

**Comment by smn** [comment]
:+1: in general but a bit worried about the head-teacher assumption.

**Pull request merged**